client/components/seo/preview-upgrade-nudge/test/index.jsx

```diff
@@ -32,57 +35,61 @@ const props = {
 
 describe( 'SeoPreviewNudge basic tests', () => {
 	test( 'should not blow up', () => {
-		const comp = shallow( <SeoPreviewNudge { ...props } /> );
-		expect( comp.find( '.preview-upgrade-nudge' ).length ).toBe( 1 );
+		const { container } = render( <SeoPreviewNudge { ...props } /> );
+		expect( container.firstChild ).toHaveClass( 'preview-upgrade-nudge' );
 	} );
 
 	test( 'should track view', () => {
-		const comp = shallow( <SeoPreviewNudge { ...props } /> );
-		expect( comp.find( 'TrackComponentView' ).length ).toBe( 1 );
-		expect( comp.find( 'TrackComponentView' ).props().eventName ).toBe(
-			'calypso_seo_preview_upgrade_nudge_impression'
-		);
+		render( <SeoPreviewNudge { ...props } /> );
+		const track = screen.queryByTestId( 'track-component-view' );
+		expect( track ).toBeVisible();
+		expect( JSON.parse( track.textContent ) ).toMatchObject( {
```

Comment:
Instead, can we use `toHaveTextContent( 'calypso_seo_preview_upgrade_nudge_impression' )` for example? Should be enough IMHO, and it will still use one of jest dom's matchers.